client/src/components/Allocation/AllocationLowUqScore/AllocationLowUqScore.tsx

```diff
@@ -1,21 +1,78 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { Trans, useTranslation } from 'react-i18next';
-import { useNavigate } from 'react-router-dom';
+import { useAccount } from 'wagmi';
 
 import BoxRounded from 'components/ui/BoxRounded';
 import Button from 'components/ui/Button';
 import InputCheckbox from 'components/ui/InputCheckbox';
-import { ROOT_ROUTES } from 'routes/RootRoutes/routes';
+import { GITCOIN_PASSPORT_CUSTOM_OCTANT_DASHBOARD } from 'constants/urls';
+import useRefreshAntisybilStatus from 'hooks/mutations/useRefreshAntisybilStatus';
+import useCurrentEpoch from 'hooks/queries/useCurrentEpoch';
+import useUqScore from 'hooks/queries/useUqScore';
+import toastService from 'services/toastService';
 
 import styles from './AllocationLowUqScore.module.scss';
 import AllocationLowUqScoreProps from './types';
 
-const AllocationLowUqScore: FC<AllocationLowUqScoreProps> = ({ onAllocate }) => {
+const AllocationLowUqScore: FC<AllocationLowUqScoreProps> = ({ onAllocate, onCloseModal }) => {
   const { t } = useTranslation('translation', {
     keyPrefix: 'components.allocation.lowUQScoreModal',
   });
   const [isChecked, setIsChecked] = useState(false);
-  const navigate = useNavigate();
+  const [isRefetchRequired, setIsRefetchRequired] = useState(false);
+
+  const { address } = useAccount();
+
+  const { data: currentEpoch } = useCurrentEpoch();
+  const {
+    data: uqScore,
+    refetch: refetchUqScore,
+    isFetching: isFetchingUqScore,
+  } = useUqScore(currentEpoch!);
+  const { mutateAsync: refreshAntisybilStatus, isPending: isPendingRefreshAntisybilStatus } =
+    useRefreshAntisybilStatus();
+
+  const windowOnFocus = () => {
+    if (!address) {
+      return;
+    }
+    refreshAntisybilStatus(address).then(() => {
+      refetchUqScore();
+      setIsRefetchRequired(false);
+    });
+  };
+
+  const windowOnBlur = () => {
+    setIsRefetchRequired(true);
+  };
+
+  useEffect(() => {
+    window.addEventListener('focus', windowOnFocus);
```

Comment:
I suggest adding listeners only when user opens `GITCOIN_PASSPORT_CUSTOM_OCTANT_DASHBOARD` link.

Reply:
I thought about that but there is a slight change user may bump their score in another window, so maybe safer with focus?